.github/actions/integration-test/action.yml

```diff
@@ -0,0 +1,48 @@
+name: 'integration-test'
+description: 'Runs an integration test and uploads the results'
+author: 'dd'
+inputs:
+  test-server:
+    description: 'Test runner server specification file'
+    required: true
+    default: 'workspace-wsmtest.json'
+  test:
+    description: 'Test suite or config to run'
+    required: true
+    default: 'suites/FullIntegration.json'
+  vault-token:
+    description: 'Vault token for rendering'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Set permissions and render test configurations
+      run: |
+        cd service
+        chmod +x render_config.sh
+        ./render_config.sh dev ${{ inputs.vault-token }}
+        chmod +r rendered
+      shell: bash
+
+    # We run the test as one script, collecting the test result
+    - name: Run the test
+      id: test-run
+      run: |
+        cd integration
```

Comment:
Yes. An alternative would have been to use a script instead of an action and invoke the script multiple times. I used the action form before I understood that it did not support calling other actions. In fact, I started with three actions and ended up with just this one. I still think this is a good form and will get better as GitHub adds use of actions.

I ended up merging the steps to be able to hold the test result for the return from the run. It started as separate steps, but it got a lot longer just passing around the testResult and resultDir. So something of a compromise.